test/parallel/test-stream-readable-async-iterators.js

```diff
@@ -115,6 +115,23 @@ async function tests() {
     readable.destroy(new Error('kaboom'));
   })();
 
+  await (async function() {
+    console.log('call next() after error');
+    const readable = new Readable({
+      read() {}
+    })
+
+    let err
+    try {
+      const iterator = readable[Symbol.asyncIterator]();
+      readable.destroy(new Error('kaboom'));
+      await iterator.next();
+    } catch (e) {
+      err = e;
+    }
+    assert.strictEqual(err.message, 'kaboom');
```

Comment:
Can we assert that we have reference equality and assert Node doesn't mutate the error (basically define the error above and assert that `strictEqual` on the instance?

Also - you can use the newly added `assert.rejects` (although no obligation)